Source/x11/XGServer.m

```diff
@@ -515,56 +523,67 @@ - (Display *) xDisplay
   return dpy;
 }
 
-- (XGScreenContext *) _screenContextForScreen: (int)screen_number
+/**
+   Returns the root window of the display 
+*/
+- (Window) xDisplayRootWindow;
 {
-  int count = ScreenCount(dpy);
-  XGScreenContext *screen;
+  return RootWindow(dpy, defScreen);
+}
 
-  if (screen_number >= count)
-    {
-      [NSException raise: NSInvalidArgumentException
-                   format: @"Request for invalid screen"];
-    }
+/**
+   Returns the application root window, which is used for many things
+   such as window hints 
+*/
+- (Window) xAppRootWindow
+{
+  return generic.appRootWindow;
+}
 
-  screen = NSMapGet(screenList, (void *)(uintptr_t)screen_number);
-  if (screen == NULL)
+- (XGScreenContext *) _defaultScreenContext
+{
+  XGScreenContext *screenCtxt;
+
+  screenCtxt = NSMapGet(screenList, (void *)(uintptr_t)defScreen);
```

Comment:
Why have a map if you only allow one screen?

Reply:
I'd prefer not to touch this until we agreed on "single Xlib screen" concept.